File: tests/r_parity/23_evalue.R
# StatsPAI E-value parity (R side) -- Module 23.
#
# Runs EValue::evalues.RR on three canonical inputs and writes the
# E-value for the point estimate plus the E-value for the CI limit.
# Tolerance: rel < 1e-6 (closed-form).

.args <- commandArgs(trailingOnly = FALSE)
.file_arg <- grep("^--file=", .args, value = TRUE)
.script_dir <- if (length(.file_arg) > 0) {
  dirname(normalizePath(sub("^--file=", "", .file_arg[1])))
} else {
  getwd()
}
source(file.path(.script_dir, "_common.R"))

suppressPackageStartupMessages({
  library(EValue)
})

MODULE <- "23_evalue"

CASES <- list(
  list(rr = 2.5, lo = 1.8, hi = 3.2, label = "moderate"),
  list(rr = 4.0, lo = 2.5, hi = 6.0, label = "strong"),
  list(rr = 1.3, lo = 1.0, hi = 1.6, label = "borderline")
)

rows <- list()
for (case in CASES) {
  ev <- EValue::evalues.RR(est = case$rr, lo = case$lo, hi = case$hi)
  # ev is a 2x3 matrix-like with rows c("RR", "E-values") and cols
  # c("point", "lower", "upper")
  ev_point <- ev["E-values", "point"]
  ev_ci    <- ev["E-values", "lower"]
  if (is.na(ev_ci)) ev_ci <- ev["E-values", "upper"]
  rows[[length(rows) + 1L]] <- parity_row(
    module = MODULE,
    statistic = paste0("evalue_est_", case$label),
    estimate = ev_point, n = 1
  )
  rows[[length(rows) + 1L]] <- parity_row(
    module = MODULE,
    statistic = paste0("evalue_ci_", case$label),
    estimate = ev_ci, n = 1
  )
}

write_results(MODULE, rows, extra = list(measure = "RR"))

File: tests/r_parity/23_evalue.py
"""StatsPAI E-value parity (Python side) -- Module 23.

E-value (VanderWeele & Ding 2017) is a closed-form sensitivity-to-
unmeasured-confounding bound. We test three canonical inputs (a
moderate RR, a strong RR, an OR converted to RR) against
EValue::evalues.RR.

Tolerance: rel < 1e-6 (closed-form formula).
"""
from __future__ import annotations

import statspai as sp

from _common import ParityRecord, write_results


MODULE = "23_evalue"


CASES = [
    # (rr, rr_lower, rr_upper, label)
    (2.5, 1.8, 3.2, "moderate"),
    (4.0, 2.5, 6.0, "strong"),
    (1.3, 1.0, 1.6, "borderline"),
]


def main() -> None:
    rows: list[ParityRecord] = []
    for rr, lo, hi, label in CASES:
        out = sp.evalue(estimate=rr, ci=(lo, hi), measure="RR")
        rows.append(ParityRecord(
            module=MODULE, side="py",
            statistic=f"evalue_est_{label}",
            estimate=float(out["evalue_estimate"]),
            n=1,
        ))
        rows.append(ParityRecord(
            module=MODULE, side="py",
            statistic=f"evalue_ci_{label}",
            estimate=float(out["evalue_ci"]),
            n=1,
        ))

    write_results(MODULE, "py", rows,
                  extra={"measure": "RR", "n_cases": len(CASES)})


if __name__ == "__main__":
    main()
